### PR TITLE
Add fleiss kappa metric

### LINKED ISSUE #918: Fleiss Kappa
## 🚀 Feature

Fleiss Kappa

[Wikipedia](https://en.wikipedia.org/wiki/Fleiss%27_kappa)

### Motivation

Fleiss Kappa is a metric of inter-rater agreement between $k$ raters. It is useful in many areas for example: combining multiple measures or ensemble methods.

### Pitch

Add Fleiss Kappa as metric. I implemented it myself a while ago, but think it might be a nice addition to torchmetrics: [https://github.com/cemde/FleissKappa](https://github.com/cemde/FleissKappa)

I am happy to give it a try, make the metric more trochmetrics-like and do a PR

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #918

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--1850.org.readthedocs.build/en/1850/

### CODE REVIEW

File: src/torchmetrics/nominal/fleiss_kappa.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    agreement probability over all raters if they were randomly assigned. If the raters are in complete agreement then
+    the score 1 is returned, if there is no agreement among the raters (other than what would be expected by chance)
+    then a score smaller than 0 is returned.
+
+    As input to ``forward`` and ``update`` the metric accepts the following input:
+
+    - ``ratings`` (:class:`~torch.Tensor`): Ratings of shape [n_samples, n_categories] or [n_samples, n_categories,
+      n_raters] depedenent on `mode`. If `mode` is `counts`, `ratings` must be integer and contain the number of raters
+      that chose each category. If `mode` is `probs`, `ratings` must be floating point and contain the probability/
+      logits that each rater chose each category.
```
Comment (Borda) [inline src/torchmetrics/nominal/fleiss_kappa.py:45]:
```suggestion
    - ``ratings`` (:class:`~torch.Tensor`): Ratings of shape ``[n_samples, n_categories]`` or ``[n_samples, n_categories,
      n_raters]`` depedenent on ``mode``. If ``mode`` is ``counts``, ``ratings`` must be integer and contain the number of raters
      that chose each category. If ``mode`` is ``probs``, ``ratings`` must be floating point and contain the probability/
      logits that each rater chose each category.
```

**Review by Borda — APPROVED** [review]

File: requirements/nominal_test.txt [hunk 2]
```diff
@@ -4,3 +4,4 @@
 pandas >1.0.0, <=2.0.2 # cannot pin version due to numpy version incompatibility
 dython <=0.7.4
 scipy >1.0.0, <1.11.0 # cannot pin version due to some version conflicts with `oldest` CI configuration
+statsmodels >= 0.14.0
```
Comment (stancld) [inline requirements/nominal_test.txt:7]: Don't we wanna add any upper bound like it's done for other dependencies?

File: src/torchmetrics/functional/nominal/fleiss_kappa.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        ratings = one_hot.sum(dim=-1)
+    elif mode == "counts" and (ratings.ndim != 2 or ratings.is_floating_point()):
+        raise ValueError(
+            "If argument ``mode`` is `counts`, ratings must have 2 dimensions with the format"
+            " [n_samples, n_categories] and be none floating point."
+        )
+    return ratings
+
+
+def _fleiss_kappa_compute(counts: Tensor) -> torch.Tensor:
```
Comment (stancld) [inline src/torchmetrics/functional/nominal/fleiss_kappa.py:44]:
```suggestion
def _fleiss_kappa_compute(counts: Tensor) -> Tensor:
```

**Review by stancld — APPROVED** [review]

File: requirements/nominal_test.txt (hunk 2 above)
Comment (SkafteNicki) [reply requirements/nominal_test.txt:7]: you are right, let me add one :]
